test/resource-limits/resource_limits.td

```diff
@@ -0,0 +1,230 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ postgres-connect name=mz_system url=postgres://mz_system:materialize@${testdrive.materialize-sql-addr-internal}
+
+! ALTER SYSTEM SET max_tables TO 42
+contains:unauthorized: only user 'mz_system' is allowed to execute 'ALTER SYSTEM ...'
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_tables = 2
+
+> SHOW max_tables
+2
+
+> CREATE TABLE t1 (a INT)
+
+> CREATE TABLE t2 (a INT)
+
+! CREATE TABLE t3 (a INT)
+contains:Table resource limit of 2 cannot be exceeded. Current amount is 2.
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_tables = 25
+
+> SHOW max_tables
+25
+
+> CREATE TABLE t3 (a INT)
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_objects_per_schema = 2
+
+> SHOW max_objects_per_schema
+2
+
+! CREATE TABLE t4 (a INT)
+contains:Objects per schema resource limit of 2 cannot be exceeded. Current amount is 3.
+
+! CREATE VIEW v as SELECT 1
+contains:Objects per schema resource limit of 2 cannot be exceeded. Current amount is 3.
+
+! CREATE INDEX ind on t1 (a)
+contains:Objects per schema resource limit of 2 cannot be exceeded. Current amount is 3.
+
+! CREATE TYPE t AS (a float8)
+contains:Objects per schema resource limit of 2 cannot be exceeded. Current amount is 3.
+
+! CREATE TABLE t4 (a INT)
+contains:Objects per schema resource limit of 2 cannot be exceeded. Current amount is 3.
+
+> CREATE SCHEMA s1;
+
+> CREATE TABLE s1.t4 (a INT)
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_objects_per_schema = 20
+
+> SHOW max_objects_per_schema
+20
+
+> CREATE VIEW v as SELECT 1
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_clusters = 3
+
+> SHOW max_clusters
+3
+
+> CREATE CLUSTER c1 REPLICAS (r1 (size '1'))
+
+> CREATE CLUSTER c2 REPLICAS (r (SIZE '1'));
+
+! CREATE CLUSTER c3 REPLICAS (r (SIZE '1'));
```

Comment:
Item No 2 . Same here, `DROP CLUSTER` should be used here to confirm that the counter decreases, allowing for new clusters to be created.

Reply:
Done